pymc3/backends/atmcmc_text.py

```diff
@@ -0,0 +1,625 @@
+"""
+Text file trace backend modified from pymc3 to work efficiently with
+ATMCMC
+
+Store sampling values as CSV files.
+
+File format
+-----------
+
+Sampling values for each chain are saved in a separate file (under a
+directory specified by the `name` argument).  The rows correspond to
+sampling iterations.  The column names consist of variable names and
+index labels.  For example, the heading
+
+  x,y__0_0,y__0_1,y__1_0,y__1_1,y__2_0,y__2_1
+
+represents two variables, x and y, where x is a scalar and y has a
+shape of (3, 2).
+"""
+from glob import glob
+
+import pickle
+import os
+import pandas as pd
+
+import pymc3 as pm
+from ..model import modelcontext
+from ..backends import base, ndarray
+from . import tracetab as ttab
+from ..blocking import DictToArrayBijection, ArrayOrdering,\
+                       ListArrayOrdering, ListToArrayBijection
+from ..step_methods.arraystep import BlockedStep
+
+try:
+    from Queue import Queue
+except:
+    from queue import Queue
+
+import multiprocessing
+import traceback
+import errno
+
+
+def worker(q_in, q_out, function, eprintignore, pshared):
+    kwargs = {}
+    if pshared is not None:
+        kwargs['pshared'] = pshared
+
+    while True:
+        i, args = q_in.get()
+        if i is None:
+            break
+
+        r, e = None, None
+        try:
+            r = function(*args, **kwargs)
+        except Exception as e:
+            if eprintignore is not None and not isinstance(e, eprintignore):
+                traceback.print_exc()
+
+        q_out.put((i, r, e))
+
+
+def parimap(function, *iterables, **kwargs):
+    """
+    More efficient Queueing compared to Pool in multiprocessing.
+    Adapted from the pyrocko package. http://www.pyrocko.org
+    """
+    assert all(
```

Comment:
assertions [can be turned off](https://wiki.python.org/moin/UsingAssertionsEffectively) when a library is run -- you might be better off manually raising an `AssertionError` or `TypeError` (or custom error!) instead.